abd-clam/src/core/cluster/tree.rs

```diff
@@ -447,4 +623,88 @@ mod tests {
         // Assert recovering was successful
         assert_trees_equal(&raw_tree, &recovered_tree);
     }
+
+    fn transpose(all_ratios: &Vec<[f64; 6]>) -> Vec<Vec<f64>> {
```

Comment:
Change the return type to `[Vec<f64>; 6]` and propagate the change to the other methods here.